### PR TITLE
fix(i18n): unsubscribe previous subscription before assigning a new subscription

### LINKED ISSUE #838: Slowdown in table component after 3.17.0
## Description

Significant performance slowdown has appeared after table component refactoring in 3.17.0. 

## Steps to reproduce the issue

1. Create table with any data. 
2. Choose 4x CPU throttling in Chrome dev tools.
3. Click on the select row checkbox many times quickly.

You should see freezes and problems with interface responsiveness. On my 2012 MacBook Pro it can cause total page freeze. Moreover, if you just place table on your page and interact with other components presented on the page, you should see same problems.

There is no such problem in 3.16.3.

Thank you.

### PR DESCRIPTION
**Pull request by Donisius**
Closes #838

### CODE REVIEW

File: src/i18n/i18n.service.ts [hunk 1]
```diff
@@ -87,6 +88,8 @@ export class Overridable {
 	 * Our base non-overridden translation.
 	 */
 	protected baseTranslation: Observable<string> = this.i18n.get(this.path);
+
+	protected subscription: Subscription;
```
Comment (cal-smith) [inline src/i18n/i18n.service.ts]: Could you add a one line comment explaining the purposes of this property?

Comment (Donisius) [reply src/i18n/i18n.service.ts]: Two lines it is

Comment (cal-smith) [reply src/i18n/i18n.service.ts]: Whatever works 😆

Comment (cal-smith) [reply src/i18n/i18n.service.ts]: Also, maybe something like 
```suggestion
	/**
	 * Subscription to the observable provided as an override (if any)
	 */
	protected subscription: Subscription;
```
?

File: src/i18n/i18n.service.ts [hunk 2]
```diff
@@ -107,9 +113,14 @@ export class Overridable {
 	 */
 	override(value: string | Observable<string>) {
 		this.isOverridden = true;
+		// To ensure that there are not multiple subscriptions created for the same observable, we
+		// unsubscribe if a subscription already exists for an observable before creating a new one.
+		if (this.subscription) {
```
Comment (zvonimirfras) [inline src/i18n/i18n.service.ts]: Shouldn't this go inside of `if (isObservable(value)) {` block so we only unsubscribe if we'll subscribe after?

File: src/i18n/i18n.service.ts [hunk 3]
```diff
@@ -107,9 +113,14 @@ export class Overridable {
...
+			this.subscription.unsubscribe();
```
Comment (zvonimirfras) [inline src/i18n/i18n.service.ts]: `subscription.unsubscribe` doesn't appear to be setting the subscription to a falsy value so maybe
```suggestion
			this.subscription.unsubscribe();
			this.subscription = null;
```
to make sure we enter here repeatedly?

File: src/i18n/i18n.service.ts (hunk 2 above)
Comment (cal-smith) [reply src/i18n/i18n.service.ts]: What happens if we have a subscription but someone passes a raw value in (for whatever reason)?

Comment (zvonimirfras) [reply src/i18n/i18n.service.ts]: Yup, makes sense.

File: src/i18n/i18n.service.ts [hunk 4]
```diff
@@ -109,7 +112,13 @@ export class Overridable {
 		this.isOverridden = true;
 		this._value = value;
 		if (isObservable(value)) {
-			value.subscribe(v => {
+			// To ensure that there are not multiple subscriptions created for the same observable, we
+			// unsubscribe if a subscription already exists for an observable before creating a new one.
+			if (this.subscription) {
```
Comment (cal-smith) [inline src/i18n/i18n.service.ts]: FYI @Donisius gotta put this back outside the `if (isObservable(value)) {`, sorry!

File: src/i18n/i18n.service.ts (hunk 1 above)
Comment (cal-smith) [reply src/i18n/i18n.service.ts]: This seems to still need done?

**Pull request merged**